tools/aa/commands/SyncCommand.java

```diff
@@ -48,11 +57,55 @@ public void run(String[] args) {
           .filter(path -> fileUtils.folderExists(path))
           .forEach(
               path -> {
-                System.out.println("Performing sync: " + path.toString());
+                System.out.println(String.format("[HEAD]: Performing sync: %s", path));
                 repoFactory.create(path).pull();
               });
     } catch (IOException e) {
       e.printStackTrace();
     }
+    // then, do the sync for all workspaces
```

Comment:
I'd like you to consider what happens if we fail at every step (even if the user pressed `ctrl-c`). Could we make it such that `ctrl-c` leaves no side effects?

For example, if we're writing to a file at location A, we could write to a file in /tmp, and then when we're done, move it from /tmp to A. This doesn't eliminate the possibility of an issue with ctrl-c, since it might prevent us from moving, but move is such a fast operation that this is unlikely to happen.

So please think about this - will ctrl-c leave us in a "bad state"?

Reply:
Let's add a TODO for this, following our discussion.